File: Python/백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys

while True:
    sides = list(map(int, sys.stdin.readline().rstrip().split(' ')))

    if sides[0] == sides[1] == sides[2] == 0:
        break

    elif sides[0] == sides[1] == sides[2]:
        print('Equilateral')

    else:
        max_side = max(sides)
        sum_side = sum(sides)

        if sum_side - max_side <= max_side:
            print("Invalid")

        else:
            sides = set(sides)

            print("Isosceles" if len(sides) == 2 else "Scalene")

File: C++17/백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    while (true) {
        int a, b, c;
        cin >> a >> b >> c;

        if (a == 0 && b == 0 && c == 0) {
            break;
        }

        if (a == b && b == c && a == c) {
            cout << "Equilateral" << "\n";
            continue;
        }

        int maxSide = max({a, b, c});

        if (a + b + c - maxSide <= maxSide) {
            cout << "Invalid" << "\n";
        } else if (a == b || b == c || a == c) {
            cout << "Isosceles" << "\n";
        } else {
            cout << "Scalene" << "\n";
        }
    }

    return 0;
}
